bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/core/mapping/operation/initializer/MultipleCompositionReferencePropertyInitializer.java

```diff
@@ -53,9 +53,10 @@ protected boolean shouldAppendExistingObjects() {
 
     @Override
     protected String inputListToIterate() {
-        return parentBusinessObject != null && withComplexMultipleInHierarchy().apply(contractInput) ? iteratorName(parentBusinessObject)
-                + "."
-                + Joiner.on(".").join(toAncestorNameListUntilMultipleComplex().apply(contractInput)) : super.inputListToIterate();
+        return contractInput.eContainer() != null && parentBusinessObject != null && withComplexMultipleInHierarchy().apply(contractInput)
```

Comment:
can you use if/else statements, I feel these statements are too long to be readable